public/app/plugins/datasource/prometheus/metric_find_query.ts

```diff
@@ -141,22 +141,35 @@ export default class PrometheusMetricFindQuery {
     const instantQuery: PromQueryRequest = { expr: query } as PromQueryRequest;
     return this.datasource.performInstantQuery(instantQuery, end).pipe(
       map((result) => {
-        return _map(result.data.data.result, (metricData) => {
-          let text = metricData.metric.__name__ || '';
-          delete metricData.metric.__name__;
-          text +=
-            '{' +
-            _map(metricData.metric, (v, k) => {
-              return k + '="' + v + '"';
-            }).join(',') +
-            '}';
-          text += ' ' + metricData.value[1] + ' ' + metricData.value[0] * 1000;
-
-          return {
-            text: text,
-            expandable: true,
-          };
-        });
+        switch (result.data.data.resultType) {
```

Comment:
https://prometheus.io/docs/prometheus/latest/querying/api/#expression-query-result-formats

Reply:
Looks good but could we add a test case for this?

Reply:
@aocenas Please check https://github.com/grafana/grafana/pull/51411/commits/6bcf1e4fb437cd1ddf9fab511b35b20979f937e3